internal/controller/eventing/eventmesh_test.go

```diff
@@ -0,0 +1,183 @@
+package eventing
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/kyma-project/kyma/components/eventing-controller/pkg/deployment"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// TestGetSecretForPublisher verifies the successful and failing retrieval
+// of secrets.
+func TestGetSecretForPublisher(t *testing.T) {
+	secretFor := func(message, namespace []byte) *corev1.Secret {
+		secret := &corev1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      deployment.PublisherName,
+				Namespace: "test-namespace",
+			},
+		}
+
+		secret.Data = make(map[string][]byte)
+
+		if len(message) > 0 {
+			secret.Data["messaging"] = message
+		}
+		if len(namespace) > 0 {
+			secret.Data["namespace"] = namespace
+		}
+
+		return secret
+	}
+
+	testCases := []struct {
+		name           string
+		messagingData  []byte
+		namespaceData  []byte
+		expectedSecret corev1.Secret
+		expectedError  error
```

Comment:
For consistency with tests in this repo, please keep the pattern:
- givenABC
- wantXYZ